### ISSUE TITLE
Incorrect InetSocketAddress construction in Http4sServlet

### ISSUE THREAD

**Issue by nigredo-tori** [http4s/http4s#2743]
https://github.com/http4s/http4s/blob/eb7f85d01c0ddfd0f0ffa2623ce5602f6a594e18/servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala#L91-L92

We create an **unresolved** (host + port) `InetSocketAddress` using `getRemoteAddr` as host. However, `getRemoteAddr` is an IP address, so the `InetSocketAddress` should be resolved! We should do something like 
```scala
new InetSocketAddress(req.getRemoteAddr, req.getRemotePort),
new InetSocketAddress(req.getLocalAddr, req.getLocalPort)
```
As far as I can see from the documentation, this will try to resolve the address, which for IP address *should* just parse the thing.

It is also possible to build an `InetAddress` with both host (`.getRemoteHost`) and IP (`.getRemoteAddress`), and create an `InetSocketAddress` from that.

**Comment by rossabaker** [comment]
This has been ported to ip4s in 0.22.  It could still be fixed in 0.21, if it's causing an itch.

**Comment by nigredo-tori** [comment]
@rossabaker, I have a workaround, and hopefully the project where I need it will get updated eventually. Thanks!
